lib/test_types.go

```diff
@@ -50,20 +50,60 @@ func (c *clientConnectionMock) Sync(ctx context.Context, in *psdbconnect.SyncReq
 }
 
 type (
-	ReadFunc             func(ctx context.Context, logger DatabaseLogger, ps PlanetScaleSource, tableName string, tc *psdbconnect.TableCursor, onResult OnResult, onCursor OnCursor) (*SerializedCursor, error)
-	CanConnectFunc       func(ctx context.Context, ps PlanetScaleSource) error
-	ListVitessShardsFunc func(ctx context.Context, ps PlanetScaleSource) ([]string, error)
+	BuildSchemaFunc     func(ctx context.Context, psc PlanetScaleSource, schemaBuilder SchemaBuilder) error
+	PingContextFunc     func(context.Context, PlanetScaleSource) error
+	GetVitessShardsFunc func(ctx context.Context, psc PlanetScaleSource) ([]string, error)
+	CloseFunc           func() error
```

Comment:
CloseFunc seems unused? I'd probably just make the TestMysqlClient type return nil on Close.